wire-ios/Wire-iOS/Sources/UserInterface/SelfProfile/QR Code/Views/QRCodeView.swift

```diff
@@ -0,0 +1,114 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import SwiftUI
+
+struct QRCodeView: View {
+
+    // MARK: - Properties
+
+    @ObservedObject var viewModel: UserQRCodeViewModel
```

Comment:
shouldn't it be a @StateObject here ?

Reply:
@jullianm @agisilaos  As far as I understand the difference between `ObservedObject` and `StateObject`, we should use `ObservedObject` since we are injecting `viewModel` and we don't need to save viewModel's state since it is being kept at higher level.

Reply:
Right, makes sense since it is injected from a higher level, I didn't see that. all good then!